lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -1440,16 +1432,27 @@ Schema Type():
 {
 
     ( SchemaProperty(props) )*
+  s = UnannotatedType(props)
```

Comment:
Added to use it for message return types (the grammar attaches annotations before the return type on the message).

Using this production here makes sure the other lookahead (in line 1444 -> 1449) is no longer needed.